ResearchKit/Common/ORKAxisView.m

```diff
@@ -0,0 +1,147 @@
+        UILabel *label = self.titleLabels[i];
+        CGFloat offset = i * segmentWidth;
+        [_variableConstraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                            attribute:NSLayoutAttributeCenterY
+                                                            relatedBy:NSLayoutRelationEqual
+                                                               toItem:label.superview
+                                                            attribute:NSLayoutAttributeCenterY
+                                                           multiplier:1.0
+                                                             constant:0.0]];
+        [_variableConstraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                            attribute:NSLayoutAttributeCenterX
+                                                            relatedBy:NSLayoutRelationEqual
+                                                               toItem:label.superview
+                                                            attribute:NSLayoutAttributeLeading
+                                                           multiplier:1.0
+                                                             constant:offset]];
+        if (i == self.titleLabels.count - 1) {
+            [_variableConstraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                                attribute:NSLayoutAttributeHeight
+                                                                relatedBy:NSLayoutRelationEqual
+                                                                   toItem:label.superview
+                                                                attribute:NSLayoutAttributeHeight
+                                                               multiplier:1.0
+                                                                 constant:-_lastLabelPadding]];
+            [_variableConstraints addObject:[NSLayoutConstraint constraintWithItem:label
+                                                                attribute:NSLayoutAttributeWidth
+                                                                relatedBy:NSLayoutRelationEqual
+                                                                   toItem:label.superview
+                                                                attribute: NSLayoutAttributeHeight
+                                                               multiplier:1.0
+                                                                 constant:-_lastLabelPadding]];
+        }
+    }
+    [NSLayoutConstraint activateConstraints:_variableConstraints];
+}
+
+- (void)setupTitles:(NSArray *)titles {
+    
+    for (NSUInteger i = 0; i < titles.count; i++) {
+        
+        UILabel *label = [UILabel new];
+        label.text = titles[i];
+        label.font = [UIFont systemFontOfSize:12.0];
```

Comment:
I added this to [Issue #241](https://github.com/ResearchKit/ResearchKit/issues/241).